### PR TITLE
New Emailer type : plain SMTP

### PR DESCRIPTION
**Pull request by zeronounours**
Add a  plain SMTP email engine.

It can be used anonymously or with an account. Gmail being just that, I merged the Emailer methods but kept a different realm entry, for logs. It could be merged too and saved directly under the same entry.

Also fix a small mistake : a missing dot in services/api/src/managers/realmmanager.js, even if I'm not sure of the impact.

### CODE REVIEW

File: services/emailer/src/emailengine.js [hunk 1]
```diff
@@ -4,17 +4,31 @@ const Mailgun = require('mailgun.js');
...
+  return await mg.messages.create(config.domain, {
+    ...email,
+    'h:Reply-To': email.replyTo,
+  });
+}
+
+async function _sendWithSmtp(config, email) {
   const { replyTo, from, to, bcc, subject, text, html, attachment } = email;
 
+  var auth = undefined;
```
Comment (camelaissani) [inline services/emailer/src/emailengine.js:19]: Could you change this line by this `let auth; `?

File: services/emailer/src/emailengine.js [hunk 2]
```diff
@@ -4,17 +4,31 @@ const Mailgun = require('mailgun.js');
...
-    port: 587,
-    secure: false, // if true then port is 465, false for other ports
-    auth: {
-      user: config.email,
-      pass: config.appPassword,
-    },
+    host: config.server,
+    port: config.ports,
+    secure: config.secure,
+    auth: auth,
```
Comment (camelaissani) [inline services/emailer/src/emailengine.js:31]: `auth,`  works as well here

File: services/api/src/managers/realmmanager.js [hunk 3]
```diff
@@ -146,7 +159,20 @@ module.exports = {
         );
       } else if (req.realm.thirdParties?.gmail?.appPassword) {
         updatedRealm.thirdParties.gmail.appPassword =
-          req.realm.thirdParties.gmailappPassword;
+          req.realm.thirdParties.gmail.appPassword;
+      }
+    }
+
+    if (updatedRealm.thirdParties?.smtp) {
+      logger.debug("realm update with smtp");
```
Comment (camelaissani) [inline services/api/src/managers/realmmanager.js:167]: WDYT to log also for the other email providers?

**Review by camelaissani — COMMENTED** [review]
Thank you and well done!

**Comment by zeronounours** [comment]
I've pushed a new commit with the pointed code improvements. JS is not the language I'm the most familiar with so it may contains some improvements like these.

I've also added debug log to other emailer.

**Review by camelaissani — APPROVED** [review]

**Comment by camelaissani** [comment]
:pray:  thank you for this great contribution!

**Pull request merged**